File: 백준/Silver/16198. 에너지 모으기/에너지 모으기.py
N = int(input())
energy = list(map(int, input().split()))

def charge(energy, cur):    
    if len(energy) == 2:
        return cur
        
    max_val = 0
    for i in range(1, len(energy)-1):
        sumone = energy[i-1] * energy[i+1]
        new_val = energy[:i] + energy[(i+1):]
        max_val = max(max_val, charge(new_val, cur + sumone))
    
    return max_val

print(charge(energy, 0))

File: 백준/Silver/16198. 에너지 모으기/에너지 모으기.js
const fs = require("fs");
// let input = fs.readFileSync("./input.txt").toString().trim().split("\n");
let input = fs.readFileSync("/dev/stdin").toString().trim().split("\n");

const N = parseInt(input[0])
const energy = input[1].split(' ').map(Number)

let ans = 0

function charge(energy, cur) {
  if (energy.length === 2) {
    ans = Math.max(ans, cur)
    return
  }

  for (let i = 1; i < energy.length - 1; i++) {
    let sumone = energy[i - 1] * energy[i + 1]
    let new_val = energy.splice(i, 1)[0]
    charge(energy, cur + sumone)
    energy.splice(i, 0, new_val)
  }
}

charge(energy, 0)
console.log(ans)